### PR TITLE
API docs for Relational assembly: Part 1

### PR DESCRIPTION
**Pull request by ajcvickers**
This covers everything up to Migrations. Query, Scaffolding, Storage, Update, and ValueGeneration still to come.

Part of issue #1148

### CODE REVIEW

File: src/EFCore.Relational/Migrations/MigrationsAnnotationProvider.cs [hunk 1]
```diff
@@ -21,20 +25,179 @@ public MigrationsAnnotationProvider([NotNull] MigrationsAnnotationProviderDepend
...
+        ///     </para>
+        /// </summary>
+        /// <param name="sequence"> The sequence. </param>
+        /// <returns> The annotations. </returns>
         public virtual IEnumerable<IAnnotation> For(ISequence sequence) => Enumerable.Empty<IAnnotation>();
 
+        /// <summary>
+        ///     <para>
+        ///         Gets provider-specific Migrations annotations for the given <see cref="IModel" />
+        ///         when it is being removed/altered.
```
Comment (bricelam) [inline src/EFCore.Relational/Migrations/MigrationsAnnotationProvider.cs]: Just removed (not altered).
(Below too.)

File: src/EFCore.Relational/Migrations/MigrationBuilder.cs [hunk 2]
```diff
@@ -12,16 +12,53 @@
 
 namespace Microsoft.EntityFrameworkCore.Migrations
 {
+    /// <summary>
+    ///     A builder providing a fluentish API for building <see cref="MigrationOperation" />s.
```
Comment (bricelam) [inline src/EFCore.Relational/Migrations/MigrationBuilder.cs]: 😆 *fluentish*

File: src/EFCore.Relational/RelationalDatabaseFacadeExtensions.cs [hunk 3]
```diff
@@ -247,24 +430,57 @@ public static IDbContextTransaction BeginTransaction([NotNull] this DatabaseFaca
...
+        ///         set on the database connection string.
+        ///     </para>
+        /// </summary>
+        /// <param name="databaseFacade"> The <see cref="DatabaseFacade" /> for the context. </param>
+        /// <param name="timeout"> The timeout to use. </param>
         public static void SetCommandTimeout([NotNull] this DatabaseFacade databaseFacade, TimeSpan timeout)
         {
             if (timeout < TimeSpan.Zero)
             {
-                throw new ArgumentException($"Timeout must be greater than or equal to zero.  Provided: {timeout.TotalSeconds} seconds.");
```
Comment (bricelam) [inline src/EFCore.Relational/RelationalDatabaseFacadeExtensions.cs]: 🙀

File: src/EFCore.Relational/Design/IAnnotationCodeGenerator.cs [hunk 4]
```diff
@@ -1,26 +1,121 @@
-﻿// Copyright ([NotNull] c) .NET Foundation. All rights reserved.
-// Licensed under the Apache License, [NotNull] Version 2.0. See License.txt in the project root for license information.
```
Comment (smitpatel) [inline src/EFCore.Relational/Design/IAnnotationCodeGenerator.cs]: R# code clean up :trollface:

File: src/EFCore.Relational/Migrations/MigrationBuilder.cs (hunk 2 above)
Comment (ajcvickers) [inline src/EFCore.Relational/Migrations/MigrationBuilder.cs]: I ignored the spell checker on this one. Really want to leave it in. :-)

Comment (bricelam) [inline src/EFCore.Relational/Migrations/MigrationBuilder.cs]: Totally fine with it; articulates my feelings about this API well

**Pull request merged**